### PR TITLE
Documentation for Multiclass SDCA

### PR DESCRIPTION
**Pull request by wschin**
Toward #2522.

### CODE REVIEW

**Comment by codemzs** [comment]
Seems good just make sure the catalog extension methods are correctly documented such as "Create <trainer>..." and "Create <trainer> with advanced options ..."

File: src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs [hunk 1]
```diff
@@ -25,9 +25,67 @@
...
+    /// use [SdcaNonCalibrated](xref:Microsoft.ML.StandardTrainersCatalog.SdcaNonCalibrated(Microsoft.ML.MulticlassClassificationCatalog.MulticlassClassificationTrainers,System.String,System.String,System.String,Microsoft.ML.Trainers.ISupportSdcaClassificationLoss,System.Nullable{System.Single},System.Nullable{System.Single},System.Nullable{System.Int32})) or
+    /// [SdcaNonCalibrated(Options)](Microsoft.ML.StandardTrainersCatalog.SdcaNonCalibrated(Microsoft.ML.MulticlassClassificationCatalog.MulticlassClassificationTrainers,Microsoft.ML.Trainers.SdcaNonCalibratedMulticlassTrainer.Options)).
+    ///
+    /// [!include[io](~/../docs/samples/docs/api-reference/io-columns-multiclass-classification.md)]
+    ///
+    /// | Output Column Name | Column Type | Description|
+    /// | -- | -- | -- |
+    /// | `Score` | array of<xref:System.Single> | The scores of all classes.Higher value means higher probability to fall into the associated class. If the i-th element has the lagest value, the predicted label index would be i.Note that i is zero-based index. |
```
Comment (singlis) [inline src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: minor - space between . and Higher

Comment (singlis) [inline src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: >lagest [](start = 178, length = 6)

largest

Comment (singlis) [inline src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: another space after .

File: src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs [hunk 2]
```diff
@@ -25,9 +25,67 @@
...
+    /// More specifically, the probability of $\boldsymbol{x}$ belonging to class $c$ is computed by $\tilde{P}(c|\boldsymbol{x}) = \frac{ e^{\hat{y}^c} }{ \sum_{c' = 1}^m e^{\hat{y}^{c'}} }$ and store at the $c$-th element in the score vector.
+    ///
+    /// ### Training Algorithm Details
+    /// The optimization algorithm is an extension of (http://jmlr.org/papers/volume14/shalev-shwartz13a/shalev-shwartz13a.pdf) following a similar path proposed in an earlier [paper](https://www.csie.ntu.edu.tw/~cjlin/papers/maxent_dual.pdf).
+    /// It is usually much faster than [L-BFGS](https://en.wikipedia.org/wiki/Limited-memory_BFGS) and [truncated Newton methods](https://en.wikipedia.org/wiki/Truncated_Newton_method) for large-scale and sparse data set.
+    ///
+    /// Regularization is a method that can render an ill-posed problem more tractable by imposing constraints that provide information to supplement the data and that prevents overfitting by penalizing model's magnitude usually measured by some norm functions.
```
Comment (singlis) [inline src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: >norm [](start = 246, length = 4)

is it worth saying normalization functions?

Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: Nop. They are independent things.

---
In reply to: [277115309](https://github.com/dotnet/machinelearning/pull/3433#discussion_r277115309) [](ancestors = 277115309)

File: src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs [hunk 3]
```diff
@@ -25,9 +25,67 @@
...
+    /// This can improve the generalization of the model learned by selecting the optimal complexity in the bias-variance tradeoff.
+    /// Regularization works by adding the penalty on the magnitude of $\boldsymbol{w}_c$, $c=1,\dots,m$ to the error of the hypothesis.
+    /// An accurate model with extreme coefficient values would be penalized more, but a less accurate model with more conservative values would be penalized less.
+    ///
+    /// This learner supports [elastic net regularization](https://en.wikipedia.org/wiki/Elastic_net_regularization): a linear combination of L1-norm (LASSO), $|| \boldsymbol{w}_c ||_1$, and L2-norm (ridge), $|| \boldsymbol{w}_c ||_2^2$ regularizations.
+    /// L1-nrom and L2-norm regularizations have different effects and uses that are complementary in certain respects.
```
Comment (natke) [inline src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: nrom -> norm

File: src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs [hunk 4]
```diff
@@ -25,9 +25,67 @@
...
+    /// | Is caching required? | No |
+    /// | Required NuGet in addition to Microsoft.ML | None |
+    ///
+    /// ### Scoring Function
+    /// This model trains linear model to solve multiclass classification problems.
+    /// Assume that the number of classes is $m$ and number of features is $n$.
+    /// It assigns the $c$-th class a coefficient vector $\boldsymbol{w}_c \in {\mathbb R}^n$ and a bias $b_c \in {\mathbb R}$, for $c=1,\dots,m$.
+    /// Given a feature vector $\boldsymbol{x} \in {\mathbb R}^n$, the $c$-th class's score would be $\hat{y}^c = \boldsymbol{w}_c^T \boldsymbol{x} + b_c$.
+    ///
+    /// If and only if the trained model is maximum entropy classifier, user can interpret the output score vector as the predicted class probabilities because [softmax function](https://en.wikipedia.org/wiki/Softmax_function) may be applied to post-process all classes' scores.
```
[the remark this answers is not in the document]
Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: Add
```
    /// If $\boldsymbol{x}$ belongs to class $c$, then $\hat{y}^c$ should be much larger than 0.
    /// In contrast, a $\hat{y}^c$ much smaller than 0 means the desired label should not be $c$.
```
when explaining the scoring function.

---
In reply to: [277018452](https://github.com/dotnet/machinelearning/pull/3433#discussion_r277018452) [](ancestors = 277018452)

File: src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs [hunk 5]
```diff
@@ -433,12 +491,46 @@ private protected override float GetInstanceWeight(FloatLabelCursor cursor)
...
+    /// ### Trainer Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Machine learning task | Multiclass classification |
+    /// | Is normalization required? | Yes |
+    /// | Is caching required? | No |
+    /// | Required NuGet in addition to Microsoft.ML | None |
+    ///
+    /// ### Scoring Function
+    /// This model trains linear model to solve multiclass classification problems.
```
[the remark this answers is not in the document]
Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs]: Thanks. Will fix another related API below.

---
In reply to: [277019106](https://github.com/dotnet/machinelearning/pull/3433#discussion_r277019106) [](ancestors = 277019106)

**Pull request merged**